modules/gapi/perf/common/gapi_render_perf_tests.cpp

```diff
@@ -0,0 +1,1238 @@
+                                Values(2),
+                                Values(LINE_8),
+                                Values(0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestMosaics, RenderTestMosaics,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(cv::Rect(100, 100, 200, 200)),
+                                Values(25),
+                                Values(0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestImages, RenderTestImages,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(cv::Rect(50, 50, 100, 100)),
+                                Values(cv::Scalar(100, 150, 60)),
+                                Values(1.0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestPolylines, RenderTestPolylines,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(std::vector<cv::Point>{{100, 100}, {200, 200}, {150, 300}, {400, 150}}),
+                                Values(cv::Scalar(100, 150, 60)),
+                                Values(2),
+                                Values(LINE_8),
+                                Values(0),
+                                Values(CV_8UC3)));
+
+INSTANTIATE_TEST_CASE_P(RenderTestPolyItems, RenderTestPolyItems,
+                        Combine(Values(szVGA, sz720p, sz1080p),
+                                Values(50),
+                                Values(50),
+                                Values(50)));
+
+
+#ifdef HAVE_FREETYPE
+PERF_TEST_P_(RenderTestFTexts, RenderFTextsPerformanceBGROCVTest)
+{
+    std::wstring text;
+    cv::Size sz;
+    cv::Point org;
+    int fh;
+    cv::Scalar color;
+    MatType type;
+
+    std::tie(text ,sz ,org ,fh ,color, type) = GetParam();
+
+    cv::Mat gapi_mat = cv::Mat(sz, type);
+    cv::Mat ref_mat(gapi_mat);
+
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::gapi::wip::draw::Prims prims;
+    prims.emplace_back(cv::gapi::wip::draw::FText{text, org, fh, color});
+
+    cv::GMat in;
+    cv::GArray<cv::gapi::wip::draw::Prim> arr;
+
+    cv::GComputation comp(cv::GIn(in, arr),
+                          cv::GOut(cv::gapi::wip::draw::render3ch(in, arr)));
+
+    // Warm-up graph engine:
+    comp.apply(gin(gapi_mat, prims), gout(gapi_mat));
+
+    TEST_CYCLE()
+    {
+        comp.apply(gin(gapi_mat, prims), gout(gapi_mat));
+    }
+
+    SANITY_CHECK_NOTHING();
+}
+
+PERF_TEST_P_(RenderTestFTexts, RenderFTextsPerformanceNV12OCVTest)
+{
+    std::wstring text;
+    cv::Size sz;
+    cv::Point org;
+    int fh;
+    cv::Scalar color;
+    MatType type;
+
+    std::tie(text ,sz ,org ,fh ,color, type) = GetParam();
+
+    cv::Mat y_ref_mat, uv_ref_mat;
+
+    cv::Mat y_in_gapi_mat(sz, CV_8UC1), uv_in_gapi_mat(sz / 2, CV_8UC2),
```

Comment:
First `Mat` you init from outside of `create_rand_mats` but second `Mat` you init inside of `create_rand_mat`.  I don't think that it's clear code. You can init both of mats inside of function or create them outside then fill using this function. It's up to you.

Reply:
Fixed